mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -115,32 +115,35 @@ class MycroftSkill:
     Args:
         name (str): skill name
         bus (MycroftWebsocketClient): Optional bus connection
-        use_settings (bool): Set to false to not use skill settings at all
+        use_settings (bool): Set to false to not use skill settings at all (DEPRECATED)
+        skill_id (str): need to be unique, by default is set from skill path
+                        but skill loader can override this
     """
-    def __init__(self, name=None, bus=None, use_settings=True):
+    def __init__(self, name=None, bus=None, use_settings=True, skill_id=""):
         self.name = name or self.__class__.__name__
         self.resting_name = None
-        self.skill_id = ''  # will be set from the path, so guaranteed unique
         self.settings_meta = None  # set when skill is loaded in SkillLoader
 
         # Get directory of skill
         #: Member variable containing the absolute path of the skill's root
         #: directory. E.g. $XDG_DATA_HOME/mycroft/skills/my-skill.me/
         self.root_dir = dirname(abspath(sys.modules[self.__module__].__file__))
+        self.skill_id = skill_id or basename(self.root_dir)
```

Comment:
Should this be allowed here? This would allow for 2 skills to advertise the same id, when would it not be determined by a class or directory name?

Reply:
with multiple skill support reporting same id means the previous skill is unloaded

this is more or less current behavior and further improved in other PRs

this is only wrong for plugin skills, which only set skill_id in _startup method

Reply:
I think Daniel's point is well taken with respect to spoofing, however, which is harder to do when you have to be a certain directory.

Reply:
I can inject bus messages directly and bypass all this, this sort of spoofing is way down the priority list. Evil skills can do this spoofing anytime in any method regardless

I will move the skill id definition completely to the startup method and make it mandatory, this should ensure it comes from skill loader 100% of the time.